mobile/src/main/java/org/openhab/habdroid/core/connection/ConnectionManagerHelper.kt

```diff
@@ -174,10 +176,13 @@ interface ConnectionManagerHelper {
     private class NetworkTypeHelperApi23 constructor(context: Context) {
         private val connectivityManager = context.getSystemService(ConnectivityManager::class.java)
         val currentConnection: ConnectionType get() {
+            if (connectivityManager == null) {
```

Comment:
As much as I dislike the `!!` operator, I wonder whether the `getSystemService` call wouldn't be a good exception.

Reply:
However the method got annotated with `@Nullable` for a reason, so I'd check for null.

Reply:
The reason is 'not every platform has every feature' (e.g. device has no Bluetooth -> `getSystemService(BLUETOOTH_SERVICE)` returns null). I don't think that will happen with connectivity manager.